bindings/rust/src/superchain.rs

```diff
@@ -0,0 +1,300 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+/// Map of superchain names to their configurations.
+pub type Superchains = HashMap<String, Superchain>;
+
+/// Map of OPChain IDs to their configurations.
+pub type OPChains = HashMap<u64, ChainConfig>;
+
+/// Map of chain IDs to their address lists.
+pub type Addresses = HashMap<u64, AddressList>;
+
+/// Map of chain IDs to their chain's genesis system configurations.
+pub type GenesisSystemConfigs = HashMap<u64, GenesisSystemConfig>;
+
+/// Map of superchain names to their implementation contract semvers.
+pub type Implementations = HashMap<String, ContractImplementations>;
+
+// TODO: should we use alloy for these?
+pub(crate) type Address = String;
+pub(crate) type Hash = String;
+
+/// A superchain configuration.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Superchain {
+    /// Superchain configuration file contents.
+    pub config: SuperchainConfig,
+    /// Chain IDs of chains that are part of this superchain.
+    pub chain_ids: Vec<u64>,
+    /// Superchain identifier, without capitalization or display changes.
+    pub superchain: String,
+}
+
+/// A superchain configuration file format
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SuperchainConfig {
+    /// Superchain name (e.g. "Mainnet")
+    pub name: String,
+    /// Superchain L1 anchor infor
+    pub l1: SuperchainL1Info,
+    /// Optional addresses for the superchain-wide default protocol versions contract.
+    pub protocol_versions_addr: Option<Address>,
+    /// Optional address for the superchain-wide default superchain config contract.
+    pub superchain_config_addr: Option<Address>,
+    /// Hardfork Configuration. These values may be overridden by individual chains.
+    #[serde(flatten)]
+    pub hardfork_defaults: HardForkConfiguration,
+}
+
+/// Superchain L1 anchor information
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SuperchainL1Info {
+    /// L1 chain ID
+    pub chain_id: u64,
+    /// L1 chain public RPC endpoint
+    pub public_rpc: String,
+    /// L1 chain explorer RPC endpoint
+    pub explorer: String,
+}
+
+/// Level of integration with the superchain.
+#[derive(Debug, Clone, Serialize_repr, Deserialize_repr)]
+#[repr(u8)]
+pub enum SuperchainLevel {
+    /// Frontier chains are chains with customizations beyond the
+    /// standard OP Stack configuration and are considered "advanced".
+    Frontier = 1,
+    /// Standard chains don't have any customizations beyond the
+    /// standard OP Stack configuration and are considered "vanilla".
+    Standard = 2,
+}
+
+/// A chain configuration.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ChainConfig {
```

Comment:
I believe alloy's ChainConfig is replicating this one in Geth: https://github.com/ethereum/go-ethereum/blob/64dccf7aa411c5c7cd36090c3d9b9892945ae813/params/config.go#L349

Seeing that you are re-defining it differently in superchain-registry's Go package here: https://github.com/ethereum-optimism/superchain-registry/blob/acf0eef75f26247d7e792ef3f661e54fa9dc73c4/superchain/superchain.go#L84

makes me think that they have been created separately voluntarily, otherwise you'd use that one in the Go package instead. I could be wrong tho!

Reply:
Agreed thanks :)